sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.PartitionProcessor.cs

```diff
@@ -187,7 +189,7 @@ public async Task ProcessEventsAsync(EventProcessorHostPartition context, IEnume
                             if (_cachedEventsBackgroundTaskCts == null && CachedEventsManager.HasCachedEvents)
                             {
                                 // If there are events waiting to be processed, and no background task running, start a monitoring cycle.
-                                _cachedEventsBackgroundTaskCts = CancellationTokenSource.CreateLinkedTokenSource(_cts.Token);
+                                _cachedEventsBackgroundTaskCts = CancellationTokenSource.CreateLinkedTokenSource(linkedCts.Token);
```

Comment:
I _think_ this was meant to use the linked token.

Reply:
Yeah, this was intentionally meant to be independent of the processing token. The _cts is what gets canceled when stop is called, so it needs to be linked to that one. But as Jesse said the processing token is only for the call to that method.